cbrkit/loaders.py

```diff
@@ -341,23 +364,54 @@ def folder(path: Path, pattern: str) -> Casebase[Any, Any] | None:
     Args:
         path: Path of the folder.
         pattern: Relative pattern for the files.
-
+        
     Returns:
         Returns a Casebase.
 
     Examples:
         >>> from pathlib import Path
+        >>> from data.cars_validation_model import Car
         >>> folder_path = Path("./data")
-        >>> result = folder(folder_path, ".csv")
+        >>> result = folder(folder_path, "*.csv")
+        >>> assert result is not None
     """
     cb: Casebase[Any, Any] = {}
 
     for file in path.glob(pattern):
         if file.is_file() and file.suffix in _single_loaders:
-            loader = _single_loaders[path.suffix]
+            loader = _single_loaders[file.suffix]
             cb[file.name] = loader(file)
 
     if len(cb) == 0:
         return None
 
     return cb
+
+
+def validate(data: dict[str, Any] | object, validation_model: BaseModel):
+    """Validates the data against a Pydantic model. Throws a ValueError if data is None or a Pydantic ValidationError if the data does not match the model.
+
+    Args:
+        data: Data to validate. Can be an entire case base or a single case.
+        validation_model: Pydantic model to validate the data.
+
+    Examples:
+        >>> from pydantic import BaseModel, PositiveInt, NonNegativeInt
+        >>> from data.cars_validation_model import Car
+        >>> from pathlib import Path
+        >>> data = path(Path("data/cars-1k.csv"))
+        >>> validate(data, Car)
+        >>> import pandas as pd
+        >>> df = pd.read_csv("data/cars-1k.csv")
+        >>> data = dataframe(df)
+        >>> validate(data, Car)
+    """
+    if data is None:
+        raise ValueError("Data is None")
+    if isinstance(data, DataFrameCasebase):
+        data = data.df.to_dict("index")
```

Comment:
This is extremely slow. Is there a way to just iterate over all Series entries instead?

Reply:
It is possible to do it in the following manner:
`for item in df.iterrows():
    validation_model(**item[1])`
However, this is much slower at 50ms vs 6ms using df.to_dict().

Reply:
Then just leave it as it is for the time being. In case someone reports performance issues we can revisit it in the future.